src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -69,9 +69,14 @@ public string UserProfileDir
 
         public Uri SwaggerEndpoint { get; set; }
 
-        public HttpState()
+        public HttpState(HttpClient httpClient)
         {
-            Client = new HttpClient();
+            if (httpClient == null)
+            {
+                throw new ArgumentNullException(nameof(httpClient));
+            }
+
+            Client = httpClient;
```

Comment:
as above, the null check can be done in one line

Reply:
Updated here:
https://github.com/aspnet/HttpRepl/commit/ff18a5d8cac490242498c8397f068261b1eb7966